dynawo/sources/Models/Modelica/Dynawo/Electrical/Machines/GeneratorPVAlphaSum.mo

```diff
@@ -0,0 +1,73 @@
+    parameter Types.ReactivePowerPu QMinPu  "Minimum reactive power in p.u (base SnRef)";
+    parameter Types.ReactivePowerPu QMaxPu  "Maximum reactive power in p.u (base SnRef)";
+    parameter Types.ActivePowerPu PAuxHVPu = 0 "Active power consumption of the high voltage auxiliary used to recalculate the P and Q limits when the auxiliaries are not explicitly modeled (base SnRef)";
+    parameter Types.ActivePowerPu QAuxHVPu = 0 "Reactive power consumption of the high voltage auxiliary used to recalculate the P and Q limits when the auxiliaries are not explicitly modeled (base SnRef)";
+    parameter Types.ActivePowerPu PAuxLVPu = 0 "Active power consumption of the low voltage auxiliary used to recalculate the P and Q limits when the auxiliaries are not explicitly modeled  (base SnRef)";
+    parameter Types.ActivePowerPu QAuxLVPu = 0 "Reactive power consumption of the low voltage auxiliary used to recalculate the P and Q limits when the auxiliaries are not explicitly modeled  (base SnRef)";
+    final parameter Types.ActivePowerPu PMinAuxPu = PMinPu - PAuxHVPu - PAuxLVPu "Minimum active power in p.u when taking into account the auxiliaries (base SnRef)";
+    final parameter Types.ActivePowerPu PMaxAuxPu = PMaxPu - PAuxHVPu - PAuxLVPu "Maximum active power in p.u when taking into account the auxiliaries (base SnRef)";
+    final parameter Types.ReactivePowerPu QMinAuxPu = QMinPu - QAuxHVPu - QAuxLVPu "Minimum reactive power in p.u when taking into account the auxiliaries (base SnRef)";
+    final parameter Types.ReactivePowerPu QMaxAuxPu = QMaxPu - QAuxHVPu - QAuxLVPu "Maximum reactive power in p.u when taking into account the auxiliaries (base SnRef)";
+
+    parameter Real AlphaRaw "Reference value of the participation of the considered generator in the frequency regulation";
+
+    Connectors.ImPin N "Signal to change the active power reference setpoint of all the generators in the system in p.u (base SnRef)";
+    Connectors.ZPin Alpha "Participation of the considered generator in the frequency regulation";
+    Connectors.ZPin AlphaSum "Sum of all the participations of all generators in the frequency regulation";
+    Connectors.ZPin URefPu (value(start = U0Pu)) "Voltage regulation set point in p.u (base UNom)";
+
+  protected
+
+    Types.ActivePowerPu PGenRawPu (start = PGen0Pu) "Active power generation without taking limits into account in p.u (base SnRef) (generator convention)";
+
+  equation
+
+    if running.value then
+      PGenRawPu = PGen0Pu + (AlphaRaw * KGover / AlphaSum.value) * N.value;
+      PGenPu = if PGenRawPu >= PMaxAuxPu then PMaxAuxPu elseif PGenRawPu <= PMinAuxPu then PMinAuxPu else PGenRawPu;
+
+      if QGenPu >= QMaxAuxPu and (pre(QGenPu)<> QMaxAuxPu or UPu < URefPu.value) then
+          QGenPu = QMaxAuxPu;
+      elseif QGenPu <= QMinAuxPu and (pre(QGenPu)<> QMinAuxPu or UPu > URefPu.value) then
+          QGenPu = QMinAuxPu;
+      else
+        UPu = URefPu.value + Lambda * (QGenPu - QGen0Pu);
```

Comment:
What kind of control have you tried to implement here? With the U+lambdaQ control there is no QGen0Pu normally speaking.

Reply:
I think it is quite the equivalent of a proportional voltage regulation (instead of a PI with UPu = URefPu.value). You are right, it is not a real U+lambdaQ. I can have a separate model for it (other than GeneratorPV, as it is not an actual PV if lambda <> 0) and change the parameter name. I propose KVoltage (similar to KGover), what do you think ?

Reply:
I got confused with the name lambda^^ Ok to rename it.
Why do you use this behavior and not a fixed voltage?

Reply:
And don't bother to implement a U+lambdaQ if you don't need it ;)

Reply:
One last point: usually we write something like this
CommandVar = CommandVar0 + K* (controlledVarRef-controlledVar)
It is the case for the governor for example:
P = P0 + k(1-omegaPu)
Here if you control the voltage maybe we should have:
Q = Q0 + k(UrefPu - UPu)?
It does't change a lot but the order of magnitude of k will be different. Do you know where you will read this value (to keep data coherency)?

Reply:
In most cases, URefPu = UPu (PI control), which means k = infinity, which is weird to implement, no ?
That is a tricky question (the data one). For now on, I don't really know. The thing is, that I needed something like this when doing a PtFige Lille (for one generator with a proportional voltage control).

Reply:
If we write U = Uref + k(Q-Q0), maybe we can use droopVoltage ? To be coherent with KGover/droop

Reply:
As far as I'm concerned, the problem for me seems to come from the fact that you try to merge two different behaviors into one model.
Why not having a proportional model (and then you don't have the infinity problem) and a proportional integral model (without the gain), with perhaps better names?

You can have base classes for the switch between statuses and so on and only change the control law in the final models.

Reply:
k = infinity is weird yes ;)
Do you think we could implement U=Uref in the model and add a control loop similar to a voltage regulator that would set Uref to URef0+ k*(QGenPu - QGen0Pu) in a preassembled?

Reply:
Similar to a reactive power control loop in fact

Reply:
I am Ok with that solution with two models. But I have no good idea for the names, besides GeneratorPVPropInt and GeneratorPVProp ?

Reply:
(sorry I didn't see Adrien's answer before giving mine, page not refreshed^^) 
These names seem fine to me

Reply:
simply GeneratorPV instead of GeneratorPVPropInt

Reply:
OK :)